File: backend/app/data/districts.py
ZIMBABWE_DISTRICTS: dict[str, dict] = {
    # Bulawayo
    "Bulawayo": {"code": "BUL", "province": "Bulawayo"},

    # Harare
    "Chitungwiza":  {"code": "CHI", "province": "Harare"},
    "Epworth":      {"code": "EPW", "province": "Harare"},
    "Harare":       {"code": "HAR", "province": "Harare"},
    "Harare Rural": {"code": "HRA", "province": "Harare"},

    # Manicaland
    "Buhera":       {"code": "BUH", "province": "Manicaland"},
    "Chimanimani":  {"code": "CHA", "province": "Manicaland"},
    "Chipinge":     {"code": "CHP", "province": "Manicaland"},
    "Makoni":       {"code": "MAK", "province": "Manicaland"},
    "Mutare":       {"code": "MUT", "province": "Manicaland"},
    "Mutare Rural": {"code": "MUR", "province": "Manicaland"},
    "Mutasa":       {"code": "MUB", "province": "Manicaland"},
    "Nyanga":       {"code": "NYA", "province": "Manicaland"},
    "Rusape":       {"code": "RUS", "province": "Manicaland"},

    # Mashonaland Central
    "Bindura":      {"code": "BIN", "province": "Mashonaland Central"},
    "Guruve":       {"code": "GUR", "province": "Mashonaland Central"},
    "Mazowe":       {"code": "MAZ", "province": "Mashonaland Central"},
    "Mbire":        {"code": "MBI", "province": "Mashonaland Central"},
    "Mount Darwin": {"code": "MOU", "province": "Mashonaland Central"},
    "Muzarabani":   {"code": "MUZ", "province": "Mashonaland Central"},
    "Mvurwi":       {"code": "MVU", "province": "Mashonaland Central"},
    "Rushinga":     {"code": "RUH", "province": "Mashonaland Central"},
    "Shamva":       {"code": "SHA", "province": "Mashonaland Central"},

    # Mashonaland East
    "Chikomba":               {"code": "CHK",  "province": "Mashonaland East"},
    "Goromonzi":              {"code": "GOR",  "province": "Mashonaland East"},
    "Hwedza":                 {"code": "HWE",  "province": "Mashonaland East"},
    "Marondera":              {"code": "MAR",  "province": "Mashonaland East"},
    "Marondera Rural":        {"code": "MRA",  "province": "Mashonaland East"},
    "Mudzi":                  {"code": "MUD",  "province": "Mashonaland East"},
    "Murehwa":                {"code": "MUA",  "province": "Mashonaland East"},
    "Mutoko":                 {"code": "MUO",  "province": "Mashonaland East"},
    "Ruwa":                   {"code": "RUW",  "province": "Mashonaland East"},
    "Seke":                   {"code": "SEK",  "province": "Mashonaland East"},
    "Uzumba-Maramba-Pfungwe": {"code": "UZUM", "province": "Mashonaland East"},
    "Wedza":                  {"code": "WED",  "province": "Mashonaland East"},

    # Mashonaland West
    "Chegutu":        {"code": "CHE", "province": "Mashonaland West"},
    "Chinhoyi":       {"code": "CHN", "province": "Mashonaland West"},
    "Hurungwe":       {"code": "HUR", "province": "Mashonaland West"},
    "Kadoma":         {"code": "KAD", "province": "Mashonaland West"},
    "Kadoma Urban":   {"code": "KAU", "province": "Mashonaland West"},
    "Kariba":         {"code": "KAR", "province": "Mashonaland West"},
    "Kariba Rural":   {"code": "KRA", "province": "Mashonaland West"},
    "Kariba Urban":   {"code": "KUA", "province": "Mashonaland West"},
    "Karoi":          {"code": "KAO", "province": "Mashonaland West"},
    "Makonde":        {"code": "MKN", "province": "Mashonaland West"},
    "Mhondoro-Ngezi": {"code": "MHO", "province": "Mashonaland West"},
    "Norton":         {"code": "NOR", "province": "Mashonaland West"},
    "Sanyati":        {"code": "SAN", "province": "Mashonaland West"},
    "Zvimba":         {"code": "ZVI", "province": "Mashonaland West"},

    # Masvingo
    "Bikita":         {"code": "BIK", "province": "Masvingo"},
    "Chiredzi":       {"code": "CHR", "province": "Masvingo"},
    "Chivi":          {"code": "CHV", "province": "Masvingo"},
    "Gutu":           {"code": "GUT", "province": "Masvingo"},
    "Masvingo":       {"code": "MAS", "province": "Masvingo"},
    "Masvingo Rural": {"code": "MSR", "province": "Masvingo"},
    "Masvingo Urban": {"code": "MSU", "province": "Masvingo"},
    "Mwenezi":        {"code": "MWE", "province": "Masvingo"},
    "Zaka":           {"code": "ZAK", "province": "Masvingo"},

    # Matabeleland North
    "Binga":          {"code": "BIA", "province": "Matabeleland North"},
    "Bubi":           {"code": "BUB", "province": "Matabeleland North"},
    "Hwange":         {"code": "HWA", "province": "Matabeleland North"},
    "Hwange Rural":   {"code": "HWR", "province": "Matabeleland North"},
    "Hwange Urban":   {"code": "HWU", "province": "Matabeleland North"},
    "Lupane":         {"code": "LUP", "province": "Matabeleland North"},
    "Nkayi":          {"code": "NKA", "province": "Matabeleland North"},
    "Tsholotsho":     {"code": "TSH", "province": "Matabeleland North"},
    "Umguza":         {"code": "UMG", "province": "Matabeleland North"},
    "Victoria Falls": {"code": "VIC", "province": "Matabeleland North"},

    # Matabeleland South
    "Beitbridge":   {"code": "BEI", "province": "Matabeleland South"},
    "Bulilima":     {"code": "BUA", "province": "Matabeleland South"},
    "Gwanda":       {"code": "GWA", "province": "Matabeleland South"},
    "Gwanda Rural": {"code": "GWR", "province": "Matabeleland South"},
    "Gwanda Urban": {"code": "GWU", "province": "Matabeleland South"},
    "Insiza":       {"code": "INS", "province": "Matabeleland South"},
    "Mangwe":       {"code": "MAN", "province": "Matabeleland South"},
    "Mangwe Rural": {"code": "MGR", "province": "Matabeleland South"},
    "Mangwe Urban": {"code": "MGU", "province": "Matabeleland South"},
    "Matobo":       {"code": "MAT", "province": "Matabeleland South"},
    "Umzingwane":   {"code": "UMZ", "province": "Matabeleland South"},

    # Midlands
    "Chirumhanzu":      {"code": "CHU", "province": "Midlands"},
    "Gokwe":            {"code": "GOK", "province": "Midlands"},
    "Gokwe Centre":     {"code": "GOC", "province": "Midlands"},
    "Gokwe North":      {"code": "GON", "province": "Midlands"},
    "Gokwe South":      {"code": "GOS", "province": "Midlands"},
    "Gokwe Urban":      {"code": "GOU", "province": "Midlands"},
    "Gweru":            {"code": "GWE", "province": "Midlands"},
    "Gweru Rural":      {"code": "GWB", "province": "Midlands"},
    "Gweru Urban":      {"code": "GWC", "province": "Midlands"},
    "Kwekwe":           {"code": "KWE", "province": "Midlands"},
    "Kwekwe Rural":     {"code": "KWR", "province": "Midlands"},
    "Kwekwe Urban":     {"code": "KWU", "province": "Midlands"},
    "Mberengwa":        {"code": "MBE", "province": "Midlands"},
    "Redcliff":         {"code": "RED", "province": "Midlands"},
    "Shurugwi":         {"code": "SHU", "province": "Midlands"},
    "Shurugwi Rural":   {"code": "SHR", "province": "Midlands"},
    "Shurugwi Urban":   {"code": "SHW", "province": "Midlands"},
    "Zvishavane":       {"code": "ZVS", "province": "Midlands"},
    "Zvishavane Rural": {"code": "ZVR", "province": "Midlands"},
    "Zvishavane Urban": {"code": "ZVU", "province": "Midlands"},
}

_LOWER_INDEX: dict[str, dict] = {k.lower(): v for k, v in ZIMBABWE_DISTRICTS.items()}


def get_district_code(district_name: str) -> str | None:
    """Return the district code for a given name, case-insensitive."""
    entry = _LOWER_INDEX.get(district_name.strip().lower())
    return entry["code"] if entry else None


def get_district_info(district_name: str) -> dict | None:
    """Return the full district dict (code, province) for a given name."""
    return _LOWER_INDEX.get(district_name.strip().lower())

File: frontend/src/data/districts.ts
export interface District {
  name: string;
  province: string;
}

export const ZIMBABWE_DISTRICTS: District[] = [
  // Bulawayo
  { name: "Bulawayo",                 province: "Bulawayo" },
  // Harare
  { name: "Chitungwiza",              province: "Harare" },
  { name: "Epworth",                  province: "Harare" },
  { name: "Harare",                   province: "Harare" },
  { name: "Harare Rural",             province: "Harare" },
  // Manicaland
  { name: "Buhera",                   province: "Manicaland" },
  { name: "Chimanimani",              province: "Manicaland" },
  { name: "Chipinge",                 province: "Manicaland" },
  { name: "Makoni",                   province: "Manicaland" },
  { name: "Mutare",                   province: "Manicaland" },
  { name: "Mutare Rural",             province: "Manicaland" },
  { name: "Mutasa",                   province: "Manicaland" },
  { name: "Nyanga",                   province: "Manicaland" },
  { name: "Rusape",                   province: "Manicaland" },
  // Mashonaland Central
  { name: "Bindura",                  province: "Mashonaland Central" },
  { name: "Guruve",                   province: "Mashonaland Central" },
  { name: "Mazowe",                   province: "Mashonaland Central" },
  { name: "Mbire",                    province: "Mashonaland Central" },
  { name: "Mount Darwin",             province: "Mashonaland Central" },
  { name: "Muzarabani",               province: "Mashonaland Central" },
  { name: "Mvurwi",                   province: "Mashonaland Central" },
  { name: "Rushinga",                 province: "Mashonaland Central" },
  { name: "Shamva",                   province: "Mashonaland Central" },
  // Mashonaland East
  { name: "Chikomba",                 province: "Mashonaland East" },
  { name: "Goromonzi",                province: "Mashonaland East" },
  { name: "Hwedza",                   province: "Mashonaland East" },
  { name: "Marondera",                province: "Mashonaland East" },
  { name: "Marondera Rural",          province: "Mashonaland East" },
  { name: "Mudzi",                    province: "Mashonaland East" },
  { name: "Murehwa",                  province: "Mashonaland East" },
  { name: "Mutoko",                   province: "Mashonaland East" },
  { name: "Ruwa",                     province: "Mashonaland East" },
  { name: "Seke",                     province: "Mashonaland East" },
  { name: "Uzumba-Maramba-Pfungwe",   province: "Mashonaland East" },
  { name: "Wedza",                    province: "Mashonaland East" },
  // Mashonaland West
  { name: "Chegutu",                  province: "Mashonaland West" },
  { name: "Chinhoyi",                 province: "Mashonaland West" },
  { name: "Hurungwe",                 province: "Mashonaland West" },
  { name: "Kadoma",                   province: "Mashonaland West" },
  { name: "Kadoma Urban",             province: "Mashonaland West" },
  { name: "Kariba",                   province: "Mashonaland West" },
  { name: "Kariba Rural",             province: "Mashonaland West" },
  { name: "Kariba Urban",             province: "Mashonaland West" },
  { name: "Karoi",                    province: "Mashonaland West" },
  { name: "Makonde",                  province: "Mashonaland West" },
  { name: "Mhondoro-Ngezi",           province: "Mashonaland West" },
  { name: "Norton",                   province: "Mashonaland West" },
  { name: "Sanyati",                  province: "Mashonaland West" },
  { name: "Zvimba",                   province: "Mashonaland West" },
  // Masvingo
  { name: "Bikita",                   province: "Masvingo" },
  { name: "Chiredzi",                 province: "Masvingo" },
  { name: "Chivi",                    province: "Masvingo" },
  { name: "Gutu",                     province: "Masvingo" },
  { name: "Masvingo",                 province: "Masvingo" },
  { name: "Masvingo Rural",           province: "Masvingo" },
  { name: "Masvingo Urban",           province: "Masvingo" },
  { name: "Mwenezi",                  province: "Masvingo" },
  { name: "Zaka",                     province: "Masvingo" },
  // Matabeleland North
  { name: "Binga",                    province: "Matabeleland North" },
  { name: "Bubi",                     province: "Matabeleland North" },
  { name: "Hwange",                   province: "Matabeleland North" },
  { name: "Hwange Rural",             province: "Matabeleland North" },
  { name: "Hwange Urban",             province: "Matabeleland North" },
  { name: "Lupane",                   province: "Matabeleland North" },
  { name: "Nkayi",                    province: "Matabeleland North" },
  { name: "Tsholotsho",               province: "Matabeleland North" },
  { name: "Umguza",                   province: "Matabeleland North" },
  { name: "Victoria Falls",           province: "Matabeleland North" },
  // Matabeleland South
  { name: "Beitbridge",               province: "Matabeleland South" },
  { name: "Bulilima",                 province: "Matabeleland South" },
  { name: "Gwanda",                   province: "Matabeleland South" },
  { name: "Gwanda Rural",             province: "Matabeleland South" },
  { name: "Gwanda Urban",             province: "Matabeleland South" },
  { name: "Insiza",                   province: "Matabeleland South" },
  { name: "Mangwe",                   province: "Matabeleland South" },
  { name: "Mangwe Rural",             province: "Matabeleland South" },
  { name: "Mangwe Urban",             province: "Matabeleland South" },
  { name: "Matobo",                   province: "Matabeleland South" },
  { name: "Umzingwane",               province: "Matabeleland South" },
  // Midlands
  { name: "Chirumhanzu",              province: "Midlands" },
  { name: "Gokwe",                    province: "Midlands" },
  { name: "Gokwe Centre",             province: "Midlands" },
  { name: "Gokwe North",              province: "Midlands" },
  { name: "Gokwe South",              province: "Midlands" },
  { name: "Gokwe Urban",              province: "Midlands" },
  { name: "Gweru",                    province: "Midlands" },
  { name: "Gweru Rural",              province: "Midlands" },
  { name: "Gweru Urban",              province: "Midlands" },
  { name: "Kwekwe",                   province: "Midlands" },
  { name: "Kwekwe Rural",             province: "Midlands" },
  { name: "Kwekwe Urban",             province: "Midlands" },
  { name: "Mberengwa",                province: "Midlands" },
  { name: "Redcliff",                 province: "Midlands" },
  { name: "Shurugwi",                 province: "Midlands" },
  { name: "Shurugwi Rural",           province: "Midlands" },
  { name: "Shurugwi Urban",           province: "Midlands" },
  { name: "Zvishavane",               province: "Midlands" },
  { name: "Zvishavane Rural",         province: "Midlands" },
  { name: "Zvishavane Urban",         province: "Midlands" },
];

export const DISTRICTS_BY_PROVINCE: Record<string, string[]> = ZIMBABWE_DISTRICTS.reduce(
  (acc, d) => {
    (acc[d.province] ??= []).push(d.name);
    return acc;
  },
  {} as Record<string, string[]>,
);

export const DISTRICT_NAMES: string[] = ZIMBABWE_DISTRICTS.map((d) => d.name);
